drake/solvers/SNOPTWrapper.cpp

```diff
@@ -27,12 +27,15 @@ using namespace Drake;
 
 // NOTE: all snopt calls will use this shared memory... so this code is NOT THREAD SAFE
 const Drake::OptimizationProblem* current_problem = NULL;
```

Comment:
I *think* the existence of this variable may be the only remaining reason why the code is still not thread safe.  and the only reason it is here is because of the callback signature for the snopt user function.  But i'm sure there is a better way to do it.